modules/compute/gke-node-pool/main.tf

```diff
@@ -367,6 +367,22 @@ resource "google_container_node_pool" "node_pool" {
       condition     = !(var.num_node_pools < 0 || var.num_slices < 0)
       error_message = "Negative integer value of num_node_pools or num_slices is not valid. Please use a positive integer value to set num_node_pools for CPUs and GPUS, and num_slices for TPUs."
     }
+    precondition {
+      condition     = var.enable_flex_start == true ? (var.auto_repair == false) : true
+      error_message = "enable_flex_start needs node auto_repair set to false."
+    }
+    precondition {
+      condition     = var.enable_flex_start == true ? (var.static_node_count == null) : true
+      error_message = "enable_flex_start does not work with static_node_count. static_node_count should be set to null."
+    }
+    precondition {
+      condition     = var.enable_flex_start == true ? (var.reservation_affinity.consume_reservation_type == "NO_RESERVATION") : true
+      error_message = "enable_flex_start only works with reservation_affinity consume_reservation_type NO_RESERVATION."
+    }
+    precondition {
+      condition     = var.enable_flex_start == true ? (var.spot == false) : true
```

Comment:
Two points on variable naming
Can we update the variables to `enable_dws_flex_start` and `enable_spot_consumption` ?

Reply:
Updating it to `enable_spot_consumption` will not be backward compatible. We have been using `spot` all along.